main/ecrecover/FNSECP256K1/mulFnSecp256k1.zkasm

```diff
@@ -9,20 +9,13 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
 ; RESOURCES:
-;   2 arith + 7 steps
+;   1 arith + 4 steps
 
 mulFnSecp256k1:
 
-        ; 1] Compute and check the multiplication over Z
-        ; A·B + 0 = [D]·2²⁵⁶ + [E]
+        ; A·B + 0 = op (mod SECP256K1_N)
         0 => C
-        $${var _mulFnSecp256k1_AB = A * B}
-        ${_mulFnSecp256k1_AB >> 256} => D
-        ${_mulFnSecp256k1_AB} => E :ARITH
+        %SECP256K1_N => D
+        ${(A*B + C) % D} => E   :ARITH_MOD
```

Comment:
optimization (-1 step):
; C changes in row after ARITH, remove C addition from executor.
${(A*B) % D} => C :ARITH_MOD, RETURN

Reply:
Solved